src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -235,7 +236,12 @@ private void BuildStarted(object sender, BuildStartedEventArgs e)
 
         _buildStartTime = e.Timestamp;
 
-        Terminal.Write(AnsiCodes.SetProgressIndeterminate);
+        // dotnet/msbuild#8958: iTerm2 treats ;9 code to post a notification instead,
+        // so disable progress reporting on Mac.
+        if (!RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
```

Comment:
Maybe abstract this check to a method named something like `TerminalSupportsProgressReporting`, which this as the current implementation?